src/pages/CardProduct.jsx

```diff
@@ -1,9 +1,11 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import '../pages-css/props.css';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { faHeart } from "@fortawesome/free-solid-svg-icons";
 import { useCart } from '../util_Contexts';
 import { useWishlist } from '../util_Contexts/wishlist-context';
+import { useAuth } from '../util_Contexts/auth-context';
```

Comment:
You should export all contexts at one place eg- index.js and then export them.

Reply:
sure i have done like that in my other project, this one i'll have to change imports in every file so i'll fix this after submission. thanks for pointing out anyways.